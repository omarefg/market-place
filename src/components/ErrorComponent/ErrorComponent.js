import React from "react";

export function ErrorComponent({ error, retry }) {
  return (
    <div>
      <p>{error}</p>
      <button onClick={retry}>Retry</button>
    </div>
  );
}
