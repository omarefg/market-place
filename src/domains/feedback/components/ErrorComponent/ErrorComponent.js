import React from "react";

export function ErrorComponent({ error, retry }) {
  return (
    <div>
      <p data-testid="ErrorComponent_title">{error}</p>
      <button data-testid="ErrorComponent_retryButton" onClick={retry}>Retry</button>
    </div>
  );
}
