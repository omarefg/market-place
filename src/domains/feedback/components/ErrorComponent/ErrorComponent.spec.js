import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { ErrorComponent } from "./ErrorComponent";

const defaultProps = {
  error: "ErrorTest",
  retry: jest.fn(),
};

const wrapper = (props = {}) => {
  const finalProps = { ...defaultProps, ...props };
  return render(<ErrorComponent {...finalProps} />);
};

describe("@domains/feedback/components/ErrorDomain", () => {
  it("Given a valid call it must render without crashing", () => {
    // Arrange
    const { container } = wrapper();
    // Assert
    expect(container).toMatchSnapshot();
  });

  it("Given a valid retry prop it mus call it when retry button is pressed", () => {
    // Arrange
    const { getByTestId } = wrapper();
    const button = getByTestId("ErrorComponent_retryButton");
    // Act
    fireEvent.click(button);
    // Assert
    expect(defaultProps.retry).toBeCalledTimes(1);
  });

  it("Given a valid error it must render the error message", () => {
    // Arrange
    const { getByTestId } = wrapper();
    const title = getByTestId("ErrorComponent_title");
    // Assert
    expect(title.innerHTML).toBe(defaultProps.error);
  });
});
