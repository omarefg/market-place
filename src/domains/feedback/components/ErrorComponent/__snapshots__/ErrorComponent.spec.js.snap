// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@domains/feedback/components/ErrorDomain Given a valid call it must render without crashing 1`] = `
<div>
  <div>
    <p
      data-testid="ErrorComponent_title"
    >
      ErrorTest
    </p>
    <button
      data-testid="ErrorComponent_retryButton"
    >
      Retry
    </button>
  </div>
</div>
`;
